src/components/anonymous-modal.jsx

```diff
@@ -20,7 +21,7 @@ const AnonymousModal = React.createClass({
     }
 
     return (
-      <BS.Modal show={showModal} container={this} onHide={onHide}>
+      <BS.Modal show={showModal} container={this} onHide={this.onHide}>
```

Comment:
surely there should be a way to avoid using `this.` in templates

Reply:
I don't understand, why should we avoid that? This is ES6 style component. Ref: https://reactjs.org/docs/components-and-props.html

Reply:
While it is possible to use `this` as a prop, it isnt desirable for normal usage where it is possible to declare the properties which are available to the template.

Reply:
I am a bit confused. Are you talking about `container={this}` or `onHide={this.onHide}`? But both of them are consistent with the react-bootstrap documentation: https://react-bootstrap.github.io/components/modal/#modals-contained

Do you mean `onHide` function should be declared in `render()` function?

Reply:
`this.onHide` is okay, it's a common pattern https://reactjs.org/docs/handling-events.html . React template isn't a template, each element is transformed into a function call, it's possible to write component without jsx, with pure function call.

I'm not sure about container prop, but I think that's part of Bootstrap.

Reply:
Ok. I defer to you on this one then.

Reply:
I note that in most other cases `this.` is not being used; this patch is removing many of them, and very rarely adding them.

Reply:
Most of the time `this.props` is changed into `props` because `React.createClass` is changed into functional component. So now `props` is being passed as an argument.
Ref: https://reactjs.org/docs/components-and-props.html#functional-and-class-components